File: 01/rust/src/part1.rs

use std::{
    fs::File,
    io::{prelude::*, BufReader},
    path::Path
};

fn lines_from_file(filename: impl AsRef<Path>) -> Vec<String> {
    let file = File::open(filename).expect("no such file");
    let buf = BufReader::new(file);
    buf.lines()
        .map(|l| l.expect("Could not parse file"))
        .collect()
}

fn main() {
    let lines = lines_from_file("input.txt");
    let mut max_cals = 0;
    let mut tmp: i32 = 0;
    for line in lines {
        if line == "" {
            if max_cals < tmp {
                max_cals = tmp;
            }
            tmp = 0;
        }
        else {
            tmp += line.parse::<i32>().unwrap();
        }
    }
    if max_cals < tmp {
        max_cals = tmp;
    }
    println!("{:?}", max_cals)
}

File: 02/python/part1.py
f = open("input.txt", "r")
score = 0
score_lookup = {"Y":2,"X":1,"Z":3,"A Y":6,"B Y":3,"C Y":0,"A X":3,"B X":0,"C X":6,"A Z":0,"B Z":6,"C Z":3}
for line in f.readlines():
    line = line.replace("\n","")
    score += score_lookup[line]
    score += score_lookup[line.split(" ")[1]]
print(score)
